### PR TITLE
Use hackage `rawlock`

### PR DESCRIPTION
**Pull request by jasagredo**
Replace the in-tree `RAWLock` with the hackage version, which is safer in presence of exceptions.

### CODE REVIEW

**Comment by jasagredo** [comment]
The `nix`-based actions will fail because hackage.nix is not yet updated.

File: ouroboros-consensus/changelog.d/js-rawlock.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Non-breaking
+
+* Use [`rawlock`](https://hackage.haskell.org/package/rawlock) instead of the in-tree implementation.
```
Comment (amesgen) [inline ouroboros-consensus/changelog.d/js-rawlock.md:3]: It is a breaking change according to PVP as we remove stuff, right?

**Review by amesgen — APPROVED** [review]
LGTM tomorrow once hackage.nix got updated :+1:

File: ouroboros-consensus/changelog.d/js-rawlock.md (hunk 1 above)
Comment (jasagredo) [reply ouroboros-consensus/changelog.d/js-rawlock.md:3]: Hmmmm. Technically yes. But it comes from the `Util` subtree 😄 . But yeah, will change that once I push the hackage.nix update too

**Pull request merged**